### PR TITLE
Fix secondary bridge

### PR DESCRIPTION
**Pull request by trozet**
Fixes a libovsdb regression where secondary bridges no longer worked, as well as adds back the CI jobs to test it.

### CODE REVIEW

**Comment by trozet** [comment]
> we can probably exclude this job: ovn-ci / e2e (control-plane, noHA, local, ipv4, snatGW, 1br)
> 
> I'll fix that in the 2nd commit.

Actually maybe not, thats the only snatGW job exercising the control plane tests.

File: go-controller/pkg/node/gateway_localnet.go [hunk 1]
```diff
@@ -19,8 +19,8 @@ import (
 	utilnet "k8s.io/utils/net"
 )
 
-func newLocalGateway(nodeName string, hostSubnets []*net.IPNet, gwNextHops []net.IP, gwIntf string,
-	gwIPs []*net.IPNet, nodeAnnotator kube.Annotator, cfg *managementPortConfig, kube kube.Interface, watchFactory factory.NodeWatchFactory) (*gateway, error) {
+func newLocalGateway(nodeName string, hostSubnets []*net.IPNet, gwNextHops []net.IP, gwIntf, egressGWIntf string, gwIPs []*net.IPNet,
```
[the remark this answers is not in the document]
Comment (tssurya) [reply go-controller/pkg/node/gateway_localnet.go:22]: oh nvm, I have a cleanup PR that has been sitting for long, will include this as well in that https://github.com/ovn-org/ovn-kubernetes/pull/2732

**Comment by squeed** [comment]
looks good to me

**Comment by trozet** [comment]
@msherif1234 some flake with the new node test cases:
Summarizing 1 Failure:

[Fail] Gateway Init Operations [It] use node retry with adding a node 
/home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/ovn/master_test.go:1396

https://github.com/ovn-org/ovn-kubernetes/runs/5993183506?check_suite_focus=true

File: go-controller/pkg/node/gateway.go [hunk 2]
```diff
@@ -180,7 +180,7 @@ func (g *gateway) Start(stopChan <-chan struct{}, wg *sync.WaitGroup) {
 	}
 }
 
-func gatewayInitInternal(nodeName, gwIntf, egressGatewayIntf string, subnets []*net.IPNet, gwNextHops []net.IP, gwIPs []*net.IPNet, nodeAnnotator kube.Annotator) (
```
Comment (tssurya) [inline go-controller/pkg/node/gateway.go:183]: I wonder why linter doesn't catch these unused variables...

File: go-controller/pkg/ovn/gateway_init.go [hunk 3]
```diff
@@ -561,7 +561,7 @@ func (oc *Controller) addExternalSwitch(prefix, interfaceID, nodeName, gatewayRo
 	// Connect GR to external_switch with mac address of external interface
 	// and that IP address. In the case of `local` gateway mode, whenever ovnkube-node container
 	// restarts a new br-local bridge will be created with a new `nicMacAddress`.
-	externalRouterPort := types.GWRouterToExtSwitchPrefix + gatewayRouter
+	externalRouterPort := prefix + types.GWRouterToExtSwitchPrefix + gatewayRouter
```
Comment (tssurya) [inline go-controller/pkg/ovn/gateway_init.go:564]: nice catch! depending on whether this was called from 1br or 2br we'd have "" or "exgw"...

**Review by tssurya — APPROVED** [review]
/lgtm! thanks Tim!

**Pull request merged**